GVFS/GVFS/CommandLine/HealthVerb.cs

```diff
@@ -67,23 +68,42 @@ protected override void Execute(GVFSEnlistment enlistment)
-
-            this.Output.WriteLine("\nTotal hydration percentage:           " + this.FormatPercent(enlistmentHealthData.PlaceholderPercentage + enlistmentHealthData.ModifiedPathsPercentage).PadLeft(longest + 7));
+            string targetDirectoryLabel = "Health of directory: " + enlistmentHealthData.TargetDirectory;
+            string totalFilesLabel = "Total files in HEAD commit:";
+            string placeholderLabel = "Files managed by VFS for Git (faster):";
+            string modifiedPathsLabel = "Files managed by Git:";
+
+            string fileLabel = "File";
+            string directoryLabel = "Dir";
+            string totalLabel = "Sum";
+
+            int labelWidth = Math.Max(Math.Max(targetDirectoryLabel.Length, totalFilesLabel.Length), Math.Max(placeholderLabel.Length, modifiedPathsLabel.Length)) + 1;
+            int fileWidth = Math.Max(enlistmentHealthData.GitTrackedFileCount.ToString("N0").Length, fileLabel.Length);
+            int directoryWidth = Math.Max(enlistmentHealthData.GitTrackedFolderCount.ToString("N0").Length, directoryLabel.Length);
+            int totalWidth = Math.Max(enlistmentHealthData.GitTrackedItemsCount.ToString("N0").Length, totalLabel.Length);
+
+            string formattedTrackedFiles = enlistmentHealthData.GitTrackedFileCount.ToString("N0").PadLeft(fileWidth);
+            string formattedTrackedFolders = enlistmentHealthData.GitTrackedFolderCount.ToString("N0").PadLeft(directoryWidth);
+            string formattedTrackedTotal = enlistmentHealthData.GitTrackedItemsCount.ToString("N0").PadLeft(totalWidth);
+            string formattedPlaceholderFiles = enlistmentHealthData.PlaceholderFileCount.ToString("N0").PadLeft(fileWidth);
+            string formattedPlaceholderFolders = enlistmentHealthData.PlaceholderFolderCount.ToString("N0").PadLeft(directoryWidth);
+            string formattedPlaceholderTotal = enlistmentHealthData.PlaceholderCount.ToString("N0").PadLeft(totalWidth);
+            string formattedPlaceholderPercentage = this.FormatPercent(enlistmentHealthData.PlaceholderPercentage);
+            string formattedModifiedPathFiles = enlistmentHealthData.ModifiedPathsFileCount.ToString("N0").PadLeft(fileWidth);
+            string formattedModifiedPathFolders = enlistmentHealthData.ModifiedPathsFolderCount.ToString("N0").PadLeft(directoryWidth);
+            string formattedModifiedPathTotal = enlistmentHealthData.ModifiedPathsCount.ToString("N0").PadLeft(totalWidth);
+            string formattedModifiedPathPercentage = this.FormatPercent(enlistmentHealthData.ModifiedPathsPercentage);
+
+            this.Output.WriteLine("\n" + targetDirectoryLabel);
+            this.Output.WriteLine(string.Empty.PadRight(labelWidth) + fileLabel.PadLeft(fileWidth) + "   " + directoryLabel.PadLeft(directoryWidth) + "   " + totalLabel.PadLeft(totalWidth));

```

Comment:
Maybe a method to generate a table row?

`string FormatRow(int fileCount, int directoryCount, int total, int fileWidth, int directoryWidth, int totalWidth)`